mlflow/types/utils.py

```diff
@@ -146,6 +160,9 @@ def _infer_schema(data: Any) -> Schema:
             "- pyspark.sql.DataFrame\n",
             "- scipy.sparse.csr_matrix\n"
             "- scipy.sparse.csc_matrix\n"
+            "- str\n"
+            "- List[str]\n"
+            "- List[Dict[str, str]]"
```

Comment:
Can we also support `Dict[str, str]` without the list wrapper? Seems like it might be a common case too.

Reply:
Added in next commit along with tests